ibm/resource_ibm_pi_capture.go

```diff
@@ -6,198 +6,221 @@ package ibm
+				Type:             schema.TypeSet,
+				Optional:         true,
+				Elem:             &schema.Schema{Type: schema.TypeString},
+				Set:              schema.HashString,
+				ForceNew:         true,
+				DiffSuppressFunc: applyOnce,
+				Description:      "List of Data volume IDs",
 			},
 
 			helpers.PIInstanceCaptureCloudStorageRegion: {
 				Type:         schema.TypeString,
 				Optional:     true,
+				ForceNew:     true,
 				Description:  "List of Regions to use",
 				ValidateFunc: validateAllowedStringValue([]string{"us-south", "us-east", "us-de"}),
 			},
 
 			helpers.PIInstanceCaptureCloudStorageAccessKey: {
 				Type:        schema.TypeString,
 				Optional:    true,
+				ForceNew:    true,
 				Description: "Name of Cloud Storage Access Key",
 			},
 			helpers.PIInstanceCaptureCloudStorageSecretKey: {
 				Type:        schema.TypeString,
 				Optional:    true,
+				ForceNew:    true,
 				Description: "Name of the Cloud Storage Secret Key",
 			},
 			helpers.PIInstanceCaptureCloudStorageImagePath: {
 				Type:        schema.TypeString,
 				Optional:    true,
-				Description: "Name of the Image Path",
+				ForceNew:    true,
+				Description: "Cloud Storage Image Path (bucket-name [/folder/../..])",
+			},
+			// Computed Attribute
+			"image_id": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Image ID of Capture Instance",
 			},
 		},
 	}
 }
 
-func resourceIBMPICaptureCreate(d *schema.ResourceData, meta interface{}) error {
+func resourceIBMPICaptureCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
 	sess, err := meta.(ClientSession).IBMPISession()
 	if err != nil {
-		return err
+		return diag.FromErr(err)
 	}
 
 	name := d.Get(helpers.PIInstanceName).(string)
 	capturename := d.Get(helpers.PIInstanceCaptureName).(string)
 	capturedestination := d.Get(helpers.PIInstanceCaptureDestination).(string)
-	powerinstanceid := d.Get(helpers.PICloudInstanceId).(string)
+	cloudInstanceID := d.Get(helpers.PICloudInstanceId).(string)
 
-	cloudstorageImagePath := d.Get(helpers.PIInstanceCaptureCloudStorageImagePath).(string)
-	if cloudstorageImagePath == "" {
-		log.Printf("CloudImagePath is not provided")
+	client := st.NewIBMPIInstanceClient(context.Background(), sess, cloudInstanceID)
 
+	captureBody := &models.PVMInstanceCapture{
+		CaptureDestination: &capturedestination,
+		CaptureName:        &capturename,
 	}
-
-	cloudstorageregion := d.Get(helpers.PIInstanceCaptureCloudStorageRegion).(string)
-	if cloudstorageregion == "" {
-		log.Printf("CloudStorageRegion is not provided")
+	if capturedestination != imageCatalogDestination {
+		if v, ok := d.GetOk(helpers.PIInstanceCaptureCloudStorageRegion); ok {
+			captureBody.CloudStorageRegion = v.(string)
+		} else {
+			return diag.Errorf("%s  is required when capture destination is %s", helpers.PIInstanceCaptureCloudStorageRegion, cloudStorageDestination)
```

Comment:
Can you change this and 3 messages below? Because these attributes are needed even when the destination is `both`.
`cloudStorageDestination`->`capturedestination`

Reply:
Done